chore: fix the format of the deprecated comment

reddaisyy opened the pull request:
<!-- Thanks for sending a PR! Before submitting:

1. If this is your first PR, check out our contribution guide here https://docs.prylabs.network/docs/contribute/contribution-guidelines
   You will then need to sign our Contributor License Agreement (CLA), which will show up as a comment from a bot in this pull request after you open it. We cannot review code without a signed CLA.
2. Please file an associated tracking issue if this pull request is non-trivial and requires context for our team to understand. All features and most bug fixes should have
   an associated issue with a design discussed and decided upon. Small bug
   fixes and documentation improvements don't need issues.
3. New features and bug fixes must have tests. Documentation may need to
   be updated. If you're unsure what to update, send the PR, and we'll discuss
   in review.
4. Note that PRs updating dependencies and new Go versions are not accepted.
   Please file an issue instead.
5. A changelog entry is required for user facing issues.
-->

**What type of PR is this?**

> Documentation


**What does this PR do? Why is it needed?**


 `gocritic` throw a warning: `deprecatedComment: use Deprecated:  (note the casing) instead of DEPRECATED:`




**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

**Acknowledgements**

- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have included a uniquely named [changelog fragment file](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md#maintaining-changelogmd).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/OffchainLabs/prysm?pullRequest=15692) <br/>All committers have signed the CLA.

prestonvanloon commented:
If we are going to introduce this kind of change, we should include some kind of github workflow or CI check to enforce it going forward. Can you add one?

reddaisyy committed:
chore: fix the format of the deprecated comment and add CI

Signed-off-by: reddaisyy <reddaisy@outlook.jp>

reddaisyy commented:
> If we are going to introduce this kind of change, we should include some kind of github workflow or CI check to enforce it going forward. Can you add one?

@prestonvanloon   Sorry for handling this so late.

 I have added a CI to detect these issues. 

Please review again.

reddaisyy commented:
Hi,  @prestonvanloon If there's anything need I to do, please feel free to let me know.

prestonvanloon commented:
@reddaisyy looks like your CI check is failing. If we are going to address this, then we should resolve it everywhere.

reddaisyy commented:
> @reddaisyy looks like your CI check is failing. If we are going to address this, then we should resolve it everywhere.

Sorry for deal so late.  

@prestonvanloon   Thanks for pointing this out. The previous CI check scanned the entire repository, but the PR only fixed one occurrence, so it still failed on beacon-chain/rpc/prysm/v1alpha1/node/server.go.

I've updated the PR on top of the latest develop branch and fixed all remaining `DEPRECATED:` comments. I also moved the enforcement into the existing Go lint workflow using a targeted `git grep` check, and added a changelog fragment.

The relevant package tests and local workflow checks pass. Could you please take another look?
